pkg/exporter/sumologicexporter/exporter.go

```diff
@@ -38,16 +39,22 @@ const (
 )
 
 type sumologicexporter struct {
-	sources             sourceFormats
-	config              *Config
-	logger              *zap.Logger
-	client              *http.Client
+	sources sourceFormats
+	config  *Config
+	host    component.Host
+	logger  *zap.Logger
+
+	clientLock sync.RWMutex
```

Comment:
Can we add a comment here and for `dataUrlsLock`, and maybe also in the get/set functions, explaining why this is necessary?

Reply:
Added a comment around dataUrlsLock and baseUrlLock in the extension